### ISSUE TITLE
Not enough context given in SyntaxError to debug problem

### ISSUE THREAD

**Issue by WeatherGod** [snakemake/snakemake#2924]
I have a fairly complicated Snakefile, and I'm trying to debug it, but I'm not getting much useful information in a particular case:

```shell
$ snakemake --lint
SyntaxError:
Only input files can be specified as functions:
None
```
which seems to be emitted from https://github.com/snakemake/snakemake/blob/3c1065c9b441928f2d390b568a7e9f854256ad41/snakemake/rules.py#L508

Any way we can add some more useful information, like the rule name in question?

**Comment by johanneskoester** [comment]
Fixed by #2924

**Comment by WeatherGod** [comment]
Thanks!
